pajbot/migration_revisions/db/0010_make_pleblist_user_id_nullable.py

```diff
@@ -0,0 +1,27 @@
+# This migration is here to correct a previous error in the 0004_unify_user_model migration, where the pleblist_song.user_id column was incorrectly marked as non-nullable
+
+
+def up(cursor, bot):
+    # This query selects the "nullable" state of the pleblist_song.user_id column
+    cursor.execute(
+        """
+    SELECT attnotnull
+    FROM pg_attribute
+    WHERE
+        attrelid = 'pleblist_song'::regclass
+    and attname = 'user_id'
+    """
+    )
+    pleblist_song_user_id_column_unnullable = cursor.fetchone()[0]
```

Comment:
Please add a comment on why this necessary and why we can't directly "ALTER TABLE" - note that the code for revision 4 was changed after it was released, and so this migration will only run for "older" databases

Reply:
Added an additional line above the query itself